airflow/executors/kubernetes_executor.py

```diff
@@ -763,13 +763,14 @@ def _change_state(self, key: TaskInstanceKey, state: str | None, pod_id: str, na
             self.kube_scheduler.patch_pod_executor_done(pod_id=pod_id, namespace=namespace)
             self.log.info("Patched pod %s in namespace %s to mark it as done", str(key), str(namespace))
 
-        if key in self.running:
+        try:
             self.running.remove(key)
-            # We do get multiple events once the pod hits a terminal state, and we only want to
+        except KeyError:
+            self.log.debug("TI key not in running, not adding to event_buffer: %s", key)
+        else:
+            # We get multiple events once the pod hits a terminal state, and we only want to
```

Comment:
i think this changes the behavior

if no error (key is in running) else will be entered

previously, if key is in running, we do not hit else. is that desired?

Reply:
OH- -- that only was in the context of the single commit.... i see that was more a reversion